dotcom-rendering/src/lib/useOnce.ts

```diff
@@ -1,11 +1,16 @@
 import { useEffect, useState } from 'react';
 
+type ArrayOfAtLeastOne<T> = [T, ...T[]];
```

Comment:
Nice! Got something very similar in a PR open at the moment, lets make sure we consolidate https://github.com/guardian/dotcom-rendering/pull/9023/files#diff-073498780dace1ab142c60bd716457f6b7d64d8f7aa5eb1ea575497e356cf432

Reply:
As mentioned in https://github.com/guardian/dotcom-rendering/pull/9176#discussion_r1363612521, adding these to `tuple.ts` would be preferable.

I have seen [`NonEmptyArray`](https://gcanti.github.io/fp-ts/modules/NonEmptyArray.ts.html) [in](https://stackoverflow.com/questions/56006111/is-it-possible-to-define-a-non-empty-array-type-in-typescript) [a](https://matiashernandez.dev/blog/post/typescript-how-to-create-a-non-empty-array-type) [few](https://package.elm-lang.org/packages/basti1302/elm-non-empty-array/latest/Array.NonEmpty) [online tutorials](https://www.youtube.com/watch?v=1FTbE9R-pKA), but never `ArrayOfAtLeastOne`.